omroneipApp/src/drvOmroneip.cpp

```diff
@@ -0,0 +1,1882 @@
+                              updateRate_(updateRate)
+{
+}
+
+static void omronExitCallback(void *pPvt)
+{
+  drvOmronEIP *pDriver = (drvOmronEIP *)pPvt;
+  pDriver->omronExiting_ = true;
+  epicsThreadSleep(1);
+}
+
+/* This function is called by the IOC load system after iocInit() or iocRun() have completed */
+static void myInitHookFunction(initHookState state)
+{
+  switch(state) {
+    case initHookAfterIocRunning:
+      iocStarted = true;
+      break;
+    default:
+      break;
+  }
+}
+
+drvOmronEIP::drvOmronEIP(const char *portName,
+                          char *gateway,
+                          char *path,
+                          char *plcType,
+                          int debugLevel)
+
+    : asynPortDriver(portName,
+                     1,                                                                                                                                                  /* maxAddr */
+                     asynInt32Mask | asynUInt32DigitalMask | asynInt64Mask | asynFloat64Mask | asynInt32ArrayMask | asynOctetMask | asynDrvUserMask | asynInt8ArrayMask, /* Interface mask */
+                     asynInt32Mask | asynUInt32DigitalMask | asynInt64Mask | asynFloat64Mask | asynInt32ArrayMask | asynOctetMask | asynInt8ArrayMask,                   /* Interrupt mask */
+                     ASYN_CANBLOCK | ASYN_MULTIDEVICE,                                                                                                                   /* asynFlags */
+                     1,                                                                                                                                                  /* Autoconnect */
+                     0,                                                                                                                                                  /* Default priority */
+                     0),                                                                                                                                                 /* Default stack size*/
+      initialized_(false),
+      omronExiting_(false)
+
+{
+  static const char *functionName = "drvOmronEIP";
+  tagConnectionString_ = "protocol=ab-eip&gateway="+(std::string)gateway+"&path="+(std::string)path+"&plc="+(std::string)plcType;
+  std::cout<<"Starting driver with connection string: "<< tagConnectionString_<<std::endl;;
+  epicsAtExit(omronExitCallback, this);
+  plc_tag_set_debug_level(debugLevel);
+  initHookRegister(myInitHookFunction);
+  asynStatus status = (asynStatus)(epicsThreadCreate("optimiseTags",
+                          epicsThreadPriorityMedium,
+                          epicsThreadGetStackSize(epicsThreadStackMedium),
+                          (EPICSTHREADFUNC)optimiseTagsC,
+                          this) == NULL);
+  if (status!=0){
+    std::cout<< "Error, Driver initiation returned asyn status:" << status << std::endl;
+  }
+  initialized_ = true;
```

Comment:
Send an epicsEvent signal here instead

Reply:
How exactly would I do that here? I could create an event in the drivers constructor which fired an initializationEvent, then each time drvUserCreate is called I would wait for this event with a timeout? I suppose I would then have to reset the event each time it is checked for in drvUserCreate? Is there any way to leave the event in a triggered state rather rather than resetting it each time? Because refiring it for the hundreds of records I have to create wouldn't make much sense.